### ISSUE TITLE
20.0.7 to 20.0.* Maximum call stack size exceeded in nx/src/tasks-runner/create-task-graph.js:227:36

### ISSUE THREAD

**Issue by gregroyal** [nrwl/nx#28856]
### Current Behavior

upgrading from 20.0.7 to any latest version causes nested tasks to have a max call stack error. This is a hard break from being able to upgrade to the latest version of nx

example

```
  "targetDefaults": {
    "sass": {
      "dependsOn": ["^sass"],
      "inputs": [
        "sassFiles",
        "^sassFiles"
      ],
      "outputs": ["{projectRoot}/public/themes/*.css"],
      "cache": true
    },
```

from the docs https://nx.dev/features/run-tasks#defining-a-task-pipeline

if i remove '^sass' i no longer get the error, but also running this task at a parent doesnt run the child tasks first

### Expected Behavior

behavior of nx before 20.0.8

### Steps to Reproduce

1. use nx 20.0.8 or later
2. have a config like this 
```
  "targetDefaults": {
    "sass": {
      "dependsOn": ["^sass"],
      "inputs": [
        "sassFiles",
        "^sassFiles"
      ],
      "outputs": ["{projectRoot}/public/themes/*.css"],
      "cache": true
    },
```
3. have a project, that depends on another project, where they both have the same commands

### Nx Report

```shell
Node           : 20.16.0
OS             : linux-x64
Native Target  : x86_64-linux
pnpm           : 8.15.3

nx (global)  : 20.0.5
nx           : 20.0.8
```

### Failure Logs

```shell

```

### Package Manager Version

pnpm -v 8.15.3

### Operating System

- [ ] macOS
- [x] Linux
- [ ] Windows
- [ ] Other (Please specify)

**Comment by xiongemi** [comment]
maybe it is related to https://github.com/nrwl/nx/issues/28788? it is fixed in https://github.com/nrwl/nx/pull/28793 in version 20.0.11. could you try version >= 20.0.11?

**Comment by gregroyal** [comment]
yes! I think that was the same issue and looks like its fixed in 20.0.11+ ty!
